src/controllers/bundler.controller.ts

```diff
@@ -0,0 +1,43 @@
+import fs from 'fs';
+import path from 'path';
+import { NextFunction, Request, Response, Router } from 'express';
+import { ProblemException } from '../exceptions/problem.exception';
+import { Controller } from '../interfaces';
+import bundler from '@asyncapi/bundler';
+import { documentValidationMiddleware } from '../middlewares/document-validation.middleware';
+
+export class BundlerController implements Controller {
+  public basepath = '/bundle';
+  private async bundle(req: Request, res: Response, next: NextFunction) {
+    const asyncapis: Array<string> = req.body.asyncapis || [req.body.asyncapi];
+    const options: any = {};
+    if (req.body.options) {
+      if (typeof req.body.options.base === 'string') {
+        options.base = req.body.options.base;
+      }
+      if (typeof req.body.options.parse === 'object') {
+        options.parse = req.body.options.parse;
+      }
+      if (typeof req.body.options.validate === 'boolean') {
+        options.validate = req.body.options.validate;
+      }
+    }
+    try {
+      const document = await bundler.bundle(asyncapis.map(asyncapi => fs.readFileSync(path.resolve(asyncapi))), { base: options.base });
```

Comment:
I am not able to figure out how to resolve this. ☹

Reply:
You operate on content sent by the user, you cannot use filesystem because these files are sent as a value anyway, not a file path. Did I make it clearer? :)